src/RazorSdk/Targets/Sdk.Razor.CurrentVersion.targets

```diff
@@ -587,6 +587,84 @@ Copyright (c) .NET Foundation. All rights reserved.
     </ItemGroup>
   </Target>
 
+  <!--
+    Target to detect presence of .razor files and set related properties for embedded ValidatableTypeAttribute support
+    This target runs after component items are resolved to ensure accurate detection
+  -->
+  <Target Name="_AddEmbeddedValidationAttribute" BeforeTargets="GenerateGlobalUsings" DependsOnTargets="ResolveRazorComponentInputs;ResolveRazorGenerateInputs">
```

Comment:
@javiercn `GenerateGlobalUsings` has condition `@(Using->Count()) != 0`. If that condition isn't met, will this target still run?

I also don't see why to not hook into `BeforeCompile;CoreCompile` instead of `GenerateGlobalUsings`

Reply:
Why does it matter.

This needs to run before `GenerateGlobalUsings` that guarantees that. Running BeforeCompile;CoreCompile doesn't guarantee the target is scheduled before the global usings are processed

Reply:
Why exactly `_AddEmbeddedValidationAttribute` needs to be run before `GenerateGlobalUsings`?